base/services/w32time/W32Time.c

```diff
@@ -0,0 +1,370 @@
+              BOOL SystemTime)
+{
+    HANDLE hToken;
+    DWORD PrevSize;
+    TOKEN_PRIVILEGES priv, previouspriv;
+    BOOL Ret = FALSE;
+
+    /*
+     * Enable the SeSystemtimePrivilege privilege
+     */
+
+    if (OpenProcessToken(GetCurrentProcess(),
+                         TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY,
+                         &hToken))
+    {
+        priv.PrivilegeCount = 1;
+        priv.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
+
+        if (LookupPrivilegeValue(NULL,
+                                  SE_SYSTEMTIME_NAME,
+                                  &priv.Privileges[0].Luid))
+        {
+            if (AdjustTokenPrivileges(hToken,
+                                      FALSE,
+                                      &priv,
+                                      sizeof(previouspriv),
+                                      &previouspriv,
+                                      &PrevSize) &&
+                GetLastError() == ERROR_SUCCESS)
+            {
+                /*
+                 * We successfully enabled it, we're permitted to change the time.
+                 * Check the second parameter for SystemTime and if TRUE set System Time.
+                 * Otherwise, if FALSE set the Local Time.
+                 * Call SetLocalTime twice to ensure correct results.
+                 */
+                if (SystemTime)
+                {
+                    Ret = SetSystemTime(lpSystemTime);
+                }
+                else
+                {
+                    Ret = SetLocalTime(lpSystemTime) &&
+                          SetLocalTime(lpSystemTime);
+                }
+
+                /*
+                 * For the sake of security, restore the previous status again
+                 */
+                if (previouspriv.PrivilegeCount > 0)
+                {
+                    AdjustTokenPrivileges(hToken,
+                                          FALSE,
+                                          &previouspriv,
+                                          0,
+                                          NULL,
+                                          0);
+                }
+            }
+        }
+        CloseHandle(hToken);
+    }
+
+    return Ret;
+}
+
+static DWORD
+GetIntervalSetting()
+{
+    HKEY hKey;
+    DWORD dwData;
+    DWORD dwSize = sizeof(DWORD);
+
+    dwData = 0;
+    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                      L"SYSTEM\\CurrentControlSet\\Services\\W32Time\\TimeProviders\\NtpClient",
+                      0,
+                      KEY_QUERY_VALUE,
+                      &hKey) == ERROR_SUCCESS)
+    {
+        RegQueryValueExW(hKey,
+                             L"SpecialPollInterval",
+                             NULL,
+                             NULL,
+                             (LPBYTE)&dwData,
+                             &dwSize);
+        RegCloseKey(hKey);
+    }
+    return dwData;
+}
+
+int set_time()
```

Comment:
Mismatch with the style of the rest of the file

Reply:
If I move the "int" above "set-time()" does this correct it?

Reply:
I means the name of the function.
Everything is: `FunctionName`, here you do `function_name`

Reply:
Thanks for the explanation. I will change it to "SetTime".